src/cache.rs

```diff
@@ -314,3 +314,27 @@ impl Cache {
         Ok(())
     }
 }
+
+/// Unit Tests for cache module
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_page_lookup_result_iter_with_patch() {
+        let lookup = PageLookupResult::with_page(PathBuf::from("test.page"))
+            .with_optional_patch(Some( PathBuf::from("test.patch") ));
+        let mut iter = lookup.paths();
+        assert_eq!(iter.next(), Some(PathBuf::from("test.page").as_path()));
+        assert_eq!(iter.next(), Some(PathBuf::from("test.patch").as_path()));
```

Comment:
can't there just be created as `Path`s directly?

Reply:
Do you mean like this?:
```rust
        let lookup = PageLookupResult {
            page_path: PathBuf::from("test.page"),
            patch_path: PathBuf::from("test.path"),
        };
```

Reply:
I think @niklasmohrin meant this:

    assert_eq!(iter.next(), Some(Path::from("test.page")));

I haven't tested whether this works. But it should probably work, it's a Path reference with 'static lifetime.

Reply:
I think @niklasmohrin meant this:

```rust
assert_eq!(iter.next(), Some(Path::from("test.page")));
```

I haven't tested whether this works. But it should probably work, it's a Path reference with 'static lifetime.

Reply:
Oh yeah `Path::new(...)` is cleaner :). Changed the test

`Path::from(..)` didn't compile when I first tried it because of something about the `sized` trait.